### PR TITLE
Add basic support for prometheus metrics

### PR DESCRIPTION
**Pull request by plars**

### CODE REVIEW

File: testflinger/views.py [hunk 1]
```diff
@@ -0,0 +1,29 @@
+# Copyright (C) 2016-2022 Canonical
```
Comment (nadzyah) [inline testflinger/views.py:1]: I guess it should contain only 2022

File: testflinger/api/v1.py [hunk 2]
```diff
@@ -26,6 +26,11 @@
 import pkg_resources
 from flask import current_app, jsonify, request, send_file
 from werkzeug.exceptions import BadRequest
+from prometheus_client import Counter
+
+
+jobs_metric = Counter("jobs", "Number of jobs", ["queue"])
+reservations_metric = Counter("reservations", "Number of reservations")
```
Comment (nadzyah) [inline testflinger/api/v1.py:33]: I'm not sure about that, but should there be something like `registry.register` after the metrics definition?

**Review by jocave — APPROVED** [review]
Looks reasonable to me for initial metrics

File: testflinger/api/v1.py (hunk 2 above)
Comment (plars) [reply testflinger/api/v1.py:33]: I think this is only needed if you are registering a custom collector

**Pull request merged**